2020-01-Winter/4_trees/README.md

```diff
@@ -94,22 +129,87 @@ Go to [Solution](#s3)   [Top](#top)
 <!-- Don't remove -->
 <a name="s1"/>
 
-### 1. SOLUTION 1 TODO :bug:
+### 1. Path Sum
 
-Source: TODO :bug:
+Source: [Leetcode]https://leetcode.com/problems/path-sum/
 
 #### Naive/Simple Solution
-
-TODO :bug:
+This implementation uses the queue for BFS iterating over the nodes.
+**Time Complexity**: O(n) where N is the number of nodes in the tree.
+
+class Solution {
+public:
+ bool hasPathSum(TreeNode* root, int sum) {
+  //gone through the tree, no path
+  if (root == NULL)
+	  return false;
+ 
+  queue <TreeNode*> qlayer;
+  qlayer.push(root);
+  while (!qlayer.empty())
+  {
+	  TreeNode *ptr= qlayer.front();
+	  //if no children
+	  if (ptr->left == NULL && ptr->right == NULL)
+	  {
+		  //value at the end is equal to sum
+		  //means there is a path from root to leaf 
+		  if (ptr->val==sum)
+			  return true;
+	  }
+	  else
+	  {
+		  //go layer by layer
+		  //the current value is added to the left child
+		  //add the node to the queue
+		  if (ptr->left != NULL)
+		  {
+			  ptr->left->val += ptr->val;
+			  qlayer.push(ptr->left);
+		  }
+		  //the current value is added to the right child
+		  if (ptr->right != NULL)
+		  {
+			  ptr->right->val += ptr->val;
+			  qlayer.push(ptr->right);
+		  }
+	  }
+	  qlayer.pop();
+  }
+  return false;
+ }
+};
 
 #### Optimal Solution
-
-TODO :bug:
-
+This implementation uses recursion  for DFS.
+**Time Complexity**: O(2^n) the branch factor is two in a binary search tree and N is the number of times the function is called.
```

Comment:
Good job thinking about complexity tradeoffs here; double-check what you want `N` to be vs what you want the total complexity to represent. I think the total complexity will represent the number of times the function is called, whereas `N` is the number of nodes that have to be processed and perform recursive calls.

```suggestion
**Time Complexity**: `O(2^n)` -- the branch factor is two in a binary search tree and N is the number of nodes in the tree.
```